Parallel simulation builds with concatenated .cpp files (v2) (#5257)
verilator/verilator: docs/guide/exe_verilator.rst

   delayed assignments.  This option should only be used when suggested by
   the developers.

> .. option:: --output-groups <files>

Reviewer: ```suggestion
.. option:: --output-groups <numfiles>
```

As opposed to people thinking it's a filename.

   Enables concatenating the output .cpp files into the given number of

>    effective output groups.  This is useful if the compiler startup overhead

Reviewer: ```suggestion
   effective output .cpp files.  This is useful if the compiler startup overhead
```

   cumulated from compiling many small files becomes unacceptable, which can
   happen in designs making extensive use of SystemVerilog classes, templates

>    or generate blocks.  Incompatible with :vlopt:`--output-split`.

Reviewer: We should allow use with output-split (as discussed), and please add a test with this.  E.g. we split so nothing is too big, then combine so not too many files.  So really small designs might get fewer files, but larger designs should end up with ~~ numfiles of equal-ish size.

Reviewer 2: Added a test copied from `t_flag_csplit`, but with `--output-groups` specific checks added. Also, switched away from `--no-skip-identical` by just checking the Makefile (faster test re-runs).

   Using :vlopt:`--output-groups` can adversely impact caching and stability
   (as in reproducibility) of compiled code.  Compilation of larger .cpp
   files also has higher memory requirements.  Too low values might result in
   swap thrashing with large designs, high values give no benefits.  The
   value should range from 2 to 20 for small to medium designs.

>    This option is off by default.

Reviewer: ```suggestion
   Default is zero, which disables this feature.
```
